src/main/kotlin/com/sourcegraph/cody/config/CodyAuthenticationManager.kt

```diff
@@ -108,6 +124,44 @@ class CodyAuthenticationManager(val project: Project) : Disposable {
     return accountTierFuture
   }
 
+  @CalledInAny
+  private fun getIsTokenInvalid(forceRefresh: Boolean): CompletableFuture<Boolean?> {
+    isTokenInvalid?.let { if (!forceRefresh) return it }
+
+    val previousIsTokenInvalid = isTokenInvalid?.getNow(null)
+    val isTokenInvalidFuture = CompletableFuture<Boolean?>()
+    isTokenInvalid = isTokenInvalidFuture
+
+    isTokenInvalidFuture.thenApply { newAuthStatus ->
+      if (previousIsTokenInvalid != newAuthStatus) {
+        publisher.afterAction(AccountSettingChangeContext(isTokenInvalidChanged = true))
+      }
+    }
+
+    val activeAccount = getActiveAccount()
+    val token = activeAccount?.let(::getTokenForAccount)
+    if (activeAccount != null && token != null) {
+      val executor =
+          SourcegraphApiRequestExecutor.Factory.instance.create(activeAccount.server, token)
+      val progressIndicator = EmptyProgressIndicator(ModalityState.NON_MODAL)
+      val submitIOTask =
+          service<ProgressManager>().submitIOTask(progressIndicator) {
+            SourcegraphApiRequests.CurrentUser(executor, progressIndicator).getDetails()
+          }
+
+      submitIOTask.exceptionally { error ->
+        if (error.cause?.message == UNAUTHORIZED_ERROR_MESSAGE) { // invalid or expired access token
```

Comment:
This is the best we can do. This call omits the agent (this is probably a good thing). The agent checks the token in the same way (https://sourcegraph.com/github.com/sourcegraph/cody/-/blob/vscode/src/services/AuthProvider.ts?L253%3A1-335%3A1).

Reply:
Nitpick, but you could replace 5 lines with one:
`isTokenInvalidFuture.complete(error.cause?.message == UNAUTHORIZED_ERROR_MESSAGE)`